docker_cli/_shell_cmds.py

```diff
@@ -0,0 +1,369 @@
+import os
+from abc import ABC, abstractmethod
+from typing import Dict, Iterable, List, Optional, Type, Union
+
+
+class PackageManager(ABC):
+    """
+    A Linux package panager
+
+    Capabilities include:
+    -   Providing string for generating installation and configuration of
+        packages on the command line via:
+    \t  -   :func:`~docker_cli.PackageManager.generate_install_command`
+    \t  -   :func:`~docker_cli.PackageManager.generate_package_command`
+    \t  -   :func:`~docker_cli.PackageManager.generate_configure_command`
+    -   Returning an associated name and filetype
+    """
+
+    @staticmethod
+    @abstractmethod
+    def generate_install_command(
+        targets: Iterable[str], *, stringify: bool = False, clean: bool = True
+    ) -> Union[List[str], str]:
+        """
+        Returns an set of commands to install the targets.
+
+        Parameters
+        ----------
+        targets : Sequence[str]
```

Comment:
There's a minor inconsistency in the type of this `targets` parameter -- the docstring says the type is `Sequence[str]` but the type annotation above says `Iterable[str]`. Based on the implementations below, I think an iterable should be fine.

```suggestion
        targets : Iterable[str]
```